src/components/EditContactForm.js

```diff
@@ -0,0 +1,80 @@
+import React, {useState} from 'react';
+import { Link } from 'react-router-dom';
+import {formatPhoneNumber} from '../util'; 
+
+const EditContactForm = (props) => {
+  const contacts = props.contacts; 
+  const selectedId = props.selectedContactId;
+
+  //contact selected for editing
+  const editContact = contacts.find(c => c.id === selectedId);
+
+  //state
+  //pre-filled with the contact that is being edited
+  const [name, setName] = useState(editContact.name);
+  const [email, setEmail] = useState(editContact.email);
+  const [phone, setPhone] = useState(editContact.phone_number); 
+  const [picture, setPicture] = useState(editContact.image_url); 
+
+  const phoneInputHandler = (e) => {
```

Comment:
It's common to name this functions with a variable that starts with "handle" - so, `handlePhoneNumChange` or something